### PR TITLE
WIP: facilitate brandi's work on bz1830946 and bz1878246

### PR DESCRIPTION
**Pull request by gabemontero**
per scrum @bmcelvee here is what I might call some "seed planting" on what the samples doc changes might center around

as you'll see from the `ED NOTE` usage, there are several places where I've put in placed holders at best for cross referencing, replacement of OpenShift, and the like that needs to occur

I'll defer to you whether you want to make edit suggestions here for my to improve this, or if you want to take this and create/update your own commits for this.

### CODE REVIEW

**Comment by bmcelvee** [comment]
Thank you, @gabemontero! I'll take a look.

**Comment by bmcelvee** [comment]
I made some edits locally and am going to attempt to push directly to this PR. Apologies for any potential weirdness from this attempt.

File: modules/samples-operator-overview.adoc [hunk 1]
```diff
@@ -57,9 +57,40 @@ Certain circumstances will result in the Samples Operator bootstrapping itself a
...
 ====
+
+*Restricted network installation*
+
+Boostrapping as `Removed` when unable to access `registry.redhat.io`, among other things, facilitates restricted network installations when the network restriction is already in place. Not installing the samples immediately when network access is restricted allows the cluster administrator more time to decide if samples are desired before submitting alerts that sample imagestream imports are failing, which it will start doing two hours after initial installation.
+
+*Restricted network installation with initial network access*
+
+Conversely, if a cluster that is intended to be restricted network or disconnected cluster is first installed while network access exists, the Samples Operator will install the content from `registry.redhat.io` since it can access it. If you want the samples operator to still bootstrap as `Removed` in order to defer samples installation until you have decided which samples are desired, set up image mirrors, and so on, then follow the instructions at https://docs.openshift.com/container-platform/4.5/installing/install_config/installing-customizing.html [ED NOTE, fix
```
Comment (bmcelvee) [inline modules/samples-operator-overview.adoc:74]: For this link, I think I can add it to the assembly following this section, but it won't be in-line. We can xref between assemblies.

File: modules/cluster-samples-operator.adoc [hunk 2]
```diff
@@ -8,44 +8,25 @@
...
+
+On initial start up, the Operator creates the default samples configuration resource to initiate the creation of the imagestreams and templates. The configuration object
+is a clsuter scoped object with the key `cluster` and type `configs.samples`.
+
+The imagestreams are the {op-system-first}-based {product-title} imagestreams pointing to images on `registry.redhat.io`. Similarly, the templates are those categorized as {product-title} templates.
+
+The Cluster Samples Operator deployment is contained within the `openshift-cluster-samples-operator` namespace. On start up, the install pull secret is used by the imagestream import logic in the internal registry and API server to authenticate with `registry.redhat.io`. An administrator can create any additional secrets in the `openshift` namespace if they change the registry used for the sample imagestreams. If created, those secrets contain the content of a container `config.json` needed to facilitate image import.
```
Comment (bmcelvee) [inline modules/cluster-samples-operator.adoc:18]: There was a placeholder here for a link to the samples-operator-overview module. Unfortunately we still can't xref between modules. But I might be able to add a link to that assembly from the assembly that this module is in (the only catch is that it's not within the standard format of this assembly).

File: modules/cluster-samples-operator.adoc [hunk 3]
```diff
@@ -8,44 +8,25 @@
...
-annotation that denotes the {product-title} version that it is compatible with.
-The Operator uses this annotation to ensure that each sample matches it's
-release version. Samples outside of its inventory are ignored, as are skipped
-samples. Modifications to any samples that are managed by the Operator are
-reverted automatically. The jenkins images are part of the image payload from
-the installation and are tagged into the imagestreams directly.
-
-The samples resource includes a finalizer, which cleans up the following upon
-its deletion:
+The Cluster Samples Operator manages the sample imagestreams and templates stored in the `openshift` namespace, and any container credentials, stored as a secret, needed for the imagestreams to import the images they reference.
```
Comment (gabemontero) [inline modules/cluster-samples-operator.adoc:11]: missed this earlier ... we do not manage credentials.

Change this to 

`The Cluster Samples Operator manages the sample imagestreams and templates stored in the `openshift` namespace`

File: modules/cluster-samples-operator.adoc (hunk 2 above)
Comment (gabemontero) [reply modules/cluster-samples-operator.adoc:18]: ok

File: modules/cluster-samples-operator.adoc (hunk 3 above)
Comment (gabemontero) [reply modules/cluster-samples-operator.adoc:11]: We then need to revisit the section on samplesRegistry override and make sure we remind them that any auth credentials need to be stored as secrets in the `openshift` namespace, as well as point them to the internal registry doc for injecting certs for the new external registries.

File: modules/cluster-samples-operator.adoc [hunk 4]
```diff
@@ -8,44 +8,25 @@
...
+
+The Cluster Samples Operator deployment is contained within the `openshift-cluster-samples-operator` namespace. On start up, the install pull secret is used by the imagestream import logic in the internal registry and API server to authenticate with `registry.redhat.io`. An administrator can create any additional secrets in the `openshift` namespace if they change the registry used for the sample imagestreams. If created, those secrets contain the content of a container `config.json` needed to facilitate image import.
+
+The image for the Cluster Samples Operator contains imagestream and template definitions for the associated {product-title} release. Each sample includes an annotation that denotes the {product-title} version that it is compatible with. The Operator uses this annotation to ensure that each sample matches it's release version. Samples outside of its inventory are ignored, as are skipped samples. Modifications to any samples that are managed by the Operator are reverted automatically. The Jenkins images are part of the image payload from the installation and are tagged into the imagestreams directly.
```
Comment (gabemontero) [inline modules/cluster-samples-operator.adoc:20]: Another item I missed before

Each sample includes ->

After samples operator creates  sample, it adds

Comment (gabemontero) [reply modules/cluster-samples-operator.adoc:20]: Another item I missed ... Modifications to any samples ...  is also not right.

I'll submit an update.

File: modules/samples-operator-overview.adoc (hunk 1 above)
Comment (gabemontero) [reply modules/samples-operator-overview.adoc:74]: ok

File: modules/samples-operator-overview.adoc [hunk 5]
```diff
@@ -57,9 +57,40 @@ Certain circumstances will result in the Samples Operator bootstrapping itself a
...
+However, if Samples Operator also detects an OpenShift Global Proxy is configured, it bypasses those checks.
 
 [IMPORTANT]
 ====
 IPv6 installations are not currently supported by link:https://registry.redhat.io[registry.redhat.io]. The Samples Operator pulls most of the sample imagestreams and images from link:https://registry.redhat.io[registry.redhat.io].
 ====
+
+*Restricted network installation*
+
+Boostrapping as `Removed` when unable to access `registry.redhat.io`, among other things, facilitates restricted network installations when the network restriction is already in place. Not installing the samples immediately when network access is restricted allows the cluster administrator more time to decide if samples are desired before submitting alerts that sample imagestream imports are failing, which it will start doing two hours after initial installation.
```
Comment (gabemontero) [inline modules/samples-operator-overview.adoc:70]: Need to clean this up some as well ....

If removed, we'll never submit alerts 

that fact is unclear to me with this current text.  I'll throw up an alternative in a bit.

**Review by gabemontero — COMMENTED** [review]
Thanks for the edits @bmcelvee 

Re-reading alerted me to both items I mised from my initial pass, as well as slight improvements to where we landed after my initial foray and you subsequent cleanup.

I'll push a new commit in a few for you to go over that attempts to address my comments.

File: modules/cluster-samples-operator.adoc (hunk 3 above)
Comment (gabemontero) [reply modules/cluster-samples-operator.adoc:11]: Looking at https://docs.openshift.com/container-platform/4.5/openshift_images/samples-operator-alt-registry.html#installation-adding-registry-pull-secret_samples-operator-alt-registry I think we are good.

**Comment by gabemontero** [comment]
OK @bmcelvee I've pushed updates to address my first set of comments earlier today.

File: modules/samples-operator-overview.adoc (hunk 1 above)
Comment (bmcelvee) [reply modules/samples-operator-overview.adoc:74]: I put in an additional resources section following this module in the assembly. It's clunky and is probably a temporary solution. If this is a common enough occurrence, I think we'd be better off making it a separate module with a complete procedure. 

I think the content in Using the Samples Operator with an alternate registry (https://docs.openshift.com/container-platform/4.5/openshift_images/samples-operator-alt-registry.html) is also related, so I included a link to that as well.

**Comment by bmcelvee** [comment]
Getting ready to push more edits. I also went through and standardized the use of Cluster Samples Operator (instead of Cluster Samples Operator vs. Samples Operator). So, there are going to be more files surfaced.

**Comment by bmcelvee** [comment]
The build is failing on a broken link. I did something wrong, and something on my end of this branch went wonky. If I can't get it sorted cleanly, then I'll scoop everything here in to a fresh PR.

File: modules/installation-restricted-network-samples.adoc [hunk 6]
```diff
@@ -105,14 +105,14 @@ This is required because the imagestream import process does not use the mirror
...
+of the Cluster Samples Operator configuration object. Or if you do not want to support
+any of the sample imagestreams, set the Cluster Samples Operator to `Removed` in the
+Cluster Samples Operator configuration object.
 +
 [NOTE]
 ====
-Any unmirrored imagestreams that are not skipped, or if the Samples Operator is
-not changed to `Removed`, will result in the Samples Operator reporting a
+Any unmirrored imagestreams that are not skipped, or if the Cluster Samples Operator is
+not changed to `Removed`, will result in the Cluster Samples Operator reporting a
```
Comment (gabemontero) [inline modules/installation-restricted-network-samples.adoc:115]: Ah shoot found another item I missed.  In 4.6 only, we backed off marking degraded with https://github.com/openshift/cluster-samples-operator/commit/ecfb9e61ee1570abccefc77fa9b085fcfb5bfbf1

And as a replacement, it issues alerts if a) imports are failing but samples operator is periodically retrying them, b) imports are failing and samples operator does not appear to be retrying them.

**Review by gabemontero — COMMENTED** [review]
Found one more item around no longer going degraded that I missed previously @bmcelvee 

Otherwise I think that is it.

**Comment by gabemontero** [comment]
bump @bmcelvee 

where are we at ?

**Comment by bmcelvee** [comment]
Fixing my branch and updating today - then I'll kick it over for peer review (and probably QE review too, though that may not be necessary)

File: modules/installation-restricted-network-samples.adoc (hunk 6 above)
Comment (bmcelvee) [reply modules/installation-restricted-network-samples.adoc:115]: Brought forward to https://github.com/openshift/openshift-docs/pull/26134/.

**Pull request closed without merging**